pkg/agent/controller/agent.go

```diff
@@ -228,94 +304,167 @@ func (a *Controller) serviceImportLister(transform func(si *mcsv1a1.ServiceImpor
-			{
-				Cluster: a.clusterID,
-			},
-		},
+	//serviceImport := a.newServiceImport(localServiceExport.Name, localServiceExport.Namespace)
+
+	//serviceImport.Spec = mcsv1a1.ServiceImportSpec{
+	//	Ports:                 []mcsv1a1.ServicePort{},
+	//	Type:                  svcType,
+	//	SessionAffinityConfig: new(corev1.SessionAffinityConfig),
+	//}
+	//
+	//serviceImport.Status = mcsv1a1.ServiceImportStatus{
+	//	Clusters: []mcsv1a1.ClusterStatus{
+	//		{
+	//			Cluster: a.clusterID,
+	//		},
+	//	},
+	//}
+
+	//if svcType == mcsv1a1.ClusterSetIP {
+	//	if a.globalnetEnabled {
+	//		ip, reason, msg := a.getGlobalIP(svc)
+	//		if ip == "" {
+	//			klog.V(log.DEBUG).Infof("Service to be exported (%s/%s) doesn't have a global IP yet", localServiceExport.Namespace, localServiceExport.Name)
+	//			// Globalnet enabled but serviceObj doesn't have globalIp yet, Update the status and requeue
+	//			a.updateExportedServiceStatus(localServiceExport.Name, localServiceExport.Namespace, corev1.ConditionFalse, reason, msg)
+	//
+	//			return nil, true
+	//		}
+	//
+	//		serviceImport.Spec.IPs = []string{ip}
+	//	} else {
+	//		serviceImport.Spec.IPs = []string{svc.Spec.ClusterIP}
+	//	}
+	//
+	//	serviceImport.Spec.Ports = a.getPortsForService(svc)
+	//	/* We also store the clusterIP in an annotation as an optimization to recover it in case the IPs are
+	//	cleared out when here's no backing Endpoint pods.
+	//	*/
+	//	serviceImport.Annotations[clusterIP] = serviceImport.Spec.IPs[0]
+	//}
+
+	a.updateExportedServiceStatus(localServiceExport.Name, localServiceExport.Namespace,
+		mcsv1a1.ServiceExportValid, corev1.ConditionFalse, "AwaitingSync",
+		"Awaiting sync of the ServiceExport to the broker")
+
+	klog.V(log.DEBUG).Infof("Returning ServiceExport: %#v", brokerServiceExport)
+
+	return brokerServiceExport, false
+}
+
+// check whether a broker's service export originates from the local cluster
+func (a *Controller) isRemoteServiceExportOwned(se *mcsv1a1.ServiceExport) bool {
+	return se.GetLabels()[lhconstants.LighthouseLabelSourceCluster] == a.clusterID
+}
+
+func (a *Controller) serviceExportDownloadTransform(obj runtime.Object, _ int, op syncer.Operation) (runtime.Object, bool) {
+	if op != syncer.Update {
+		// we only care about status updates
+		return nil, false
 	}
 
-	if svcType == mcsv1a1.ClusterSetIP {
-		if a.globalnetEnabled {
-			ip, reason, msg := a.getGlobalIP(svc)
-			if ip == "" {
-				klog.V(log.DEBUG).Infof("Service to be exported (%s/%s) doesn't have a global IP yet", svcExport.Namespace, svcExport.Name)
-				// Globalnet enabled but service doesn't have globalIp yet, Update the status and requeue
-				a.updateExportedServiceStatus(svcExport.Name, svcExport.Namespace, corev1.ConditionFalse, reason, msg)
+	brokerServiceExport := obj.(*mcsv1a1.ServiceExport)
 
-				return nil, true
-			}
+	// we only care about service exports originating from the local cluster
+	if !a.isRemoteServiceExportOwned(brokerServiceExport) {
```

Comment:
You *may* be able to use a ListOptions if the Syncer framework allows for it when creating the Syncer, with a label match for "my cluster" (see example in pkg/mcs/lighthouse.go)